src/Ratchet/Session/Storage/Proxy/VirtualProxy.php

```diff
@@ -1,54 +1,16 @@
 <?php
-namespace Ratchet\Session\Storage\Proxy;
-use Symfony\Component\HttpFoundation\Session\Storage\Proxy\SessionHandlerProxy;
 
-class VirtualProxy extends SessionHandlerProxy {
-    /**
-     * @var string
-     */
-    protected $_sessionId;
-
-    /**
-     * @var string
-     */
-    protected $_sessionName;
-
-    /**
-     * {@inheritdoc}
-     */
-    public function __construct(\SessionHandlerInterface $handler) {
-        parent::__construct($handler);
-
-        $this->saveHandlerName = 'user';
-        $this->_sessionName    = ini_get('session.name');
-    }
-
-    /**
-     * {@inheritdoc}
-     */
-    public function getId() {
-        return $this->_sessionId;
-    }
-
-    /**
-     * {@inheritdoc}
-     */
-    public function setId($id) {
-        $this->_sessionId = $id;
-    }
-
-    /**
-     * {@inheritdoc}
-     */
-    public function getName() {
-        return $this->_sessionName;
-    }
-
-    /**
-     * DO NOT CALL THIS METHOD
-     * @internal
-     */
-    public function setName($name) {
-        throw new \RuntimeException("Can not change session name in VirtualProxy");
-    }
+/**
+ * This is a nasty workaround to support symfony/http-foundation v6+ while maintaining PHP <= 5.6 support.
+ * PHP 8 switched signature mismatches from deprecated to fatal error, causing two functions
+ * (getId and getName) to fail tests. We can't add return types with PHP versions earlier than 7.0
+ * so two versions of the VirtualProxy class need to be maintained.
+ */
+
+if (version_compare(\Composer\InstalledVersions::getVersion('symfony/http-foundation'), '6.0.0') === -1) {
+    // The version of http-foundation is < 6, include a class without return types
+    include 'VirtualProxy-HttpFoundation<6.php';
```

Comment:
This is unnecessary, the return types can be added without breaking the signature, it's removing or changing them that's an issue

Reply:
This is required if support for PHP < 7.0 is to be maintained, earlier versions didn't support return types.

Reply:
Why keep support for such old php? that's asking for maintenance hell, they have been EOL for more than 6 years

Reply:
I agree, but like I told the commenter above, it's not my project, so all I'm doing is getting it working again for php 8+

Personally, I would do it as a new major version that supported php >= 7.4 and leave < 7.4 using the current version but that's nothing to do with me.

Also, this pull request has been sitting here for a year now, I'm not sure how likely it is to even get merged at this point to be honest.

Reply:
There is ongoing work recently on ratchet from the maintainers to get the projects back alive

See

https://github.com/ratchetphp/Ratchet/issues/1054

So it's quite likely

It's okay if it's not your project, you can do a PR with a breaking change and mark it as such so it will be released as a major, otherwise those kind of big PR's are a maintenance pain and it's actually more likely to be accepted if you don't add breaking changes in your PR but drop support for older php versions

But since you made a compat layer already, you can leave it in, to be removed on the next major